test/regress_http.c

```diff
@@ -4604,6 +4604,127 @@ http_request_extra_body_test(void *arg)
+
+	tt_int_op(state->connections_done, <=, state->connections_started);
+	if (state->connections_good + state->connections_error == state->connections_started) {
+		tt_int_op(state->connections_throttled, ==, state->connections_error);
+		tt_int_op(state->connections_good + state->connections_error, ==, state->connections_done);
+		event_base_loopexit(exit_base, NULL);
+	}
+
+	return;
+end:
+	tt_fail();
+	exit(17);
+}
+
+
+static void
+http_request_done_throttlereqcb(struct evhttp_request *req, void *arg)
+{
+	struct http_throttlereqcb_test_state* state = arg;
+	if (req && evhttp_request_get_response_code(req) == HTTP_OK) {
+		state->connections_good ++;
+		evhttp_request_set_error_cb(req, NULL);
+	}
+	state->connections_done ++;
+
+	http_throttlereqcb_test_state_check(state);
+}
+
+static void
+http_request_error_throttlereqcb(enum evhttp_request_error err, void *arg)
+{
+	struct http_throttlereqcb_test_state* state = arg;
+	state->connections_error ++;
+
+	http_throttlereqcb_test_state_check(state);
+}
+
+static int
+http_throttlereqcb(struct evhttp_request* req, void *arg) {
+	struct http_throttlereqcb_test_state* state = arg;
+	state->connections_noticed ++;
+	http_throttlereqcb_test_state_check(state);
+	if (1 == state->connections_noticed%7) {
+		state->connections_throttled++;
+		return -1;
+	}
+	return 0;
+}
+
+
+static void
+http_throttlereqcb_test(void *arg)
+{
+	struct basic_test_data *data = arg;
+	ev_uint16_t port = 0;
+	struct evhttp *http = http_setup(&port, data->base, 0);
+	struct evhttp_connection *evcons[100];
+	struct http_throttlereqcb_test_state throttlereqcb_test_state;
+	unsigned n;
+
+	exit_base = data->base;
+	test_ok = 0;
+
+	memset(&throttlereqcb_test_state, 0, sizeof(throttlereqcb_test_state));
+	memset(evcons, 0, sizeof(evcons));
+
+	evhttp_set_throttlereqcb(http, http_throttlereqcb, &throttlereqcb_test_state);
+
+	for (n = 0; n < sizeof(evcons)/sizeof(evcons[0]); ++n) {
+		struct evhttp_connection* evcon = NULL;
+		struct evhttp_request *req = NULL;
+		evcons[n] = evhttp_connection_base_new(data->base, NULL, "127.0.0.1", port);
+		evcon = evcons[n];
+		evhttp_connection_set_retries(evcon, 0);
+
+		tt_assert(evcon);
+
+		req = evhttp_request_new(http_request_done_throttlereqcb, &throttlereqcb_test_state);
+		evhttp_add_header(evhttp_request_get_output_headers(req), "Connection", "close");
+		evhttp_request_set_error_cb(req, http_request_error_throttlereqcb);
+
+		/* We give ownership of the request to the connection */
+		if (evhttp_make_request(evcon, req, EVHTTP_REQ_GET, "/test") == -1) {
+			tt_abort_msg("Couldn't make request");
+		}
+
+		throttlereqcb_test_state.connections_started++;
+		http_throttlereqcb_test_state_check(&throttlereqcb_test_state);
+	}
+
+	event_base_dispatch(data->base);
+
+	http_throttlereqcb_test_state_check(&throttlereqcb_test_state);
+	tt_int_op(throttlereqcb_test_state.connections_throttled, >, 0);
+
+ end:
+	evhttp_free(http);
+
+	for (n = 0; n < sizeof(evcons)/sizeof(evcons[0]); ++n) {
```

Comment:
there is `ARRAY_SIZE()` in `regress_dnc.c`, could be moved into some header and re-used here

Reply:
would that really be clearer? People unfamiliar with the project would have to understand which kind of size is being referred to by that macro

Reply:
This is pretty common macro I suppose, but I'm fine with that sizeof/sizeof